gender_novels/analysis/gender_pronoun_freq_analysis.py

```diff
@@ -372,6 +386,156 @@ def sort_every_year(frequency_dict):
 
     return every_year_dict
 
+
+def box_gender_pronoun_freq(freq_dict, my_pal, title, x="N/A"):
+    """
+    Takes in a frequency dictionaries and exports its values as a bar-and-whisker graph
+    :param freq_dict: dictionary of frequencies grouped up
+    :param my_pal: palette to be used
+    :param title: title of exported graph
+    :param x: name of x-vars
+    :return:
+    """
+
+    plt.clf()
+    groups = []
+    val = []
+    for k, v in freq_dict.items():
+        temp = [k]*len(v)
+        groups.extend(temp)
+        val.extend(v)
+
+    df = pd.DataFrame({x: groups, 'Frequency': val})
+    df = df[[x, 'Frequency']]
+    sns.boxplot(x=df[x], y=df['Frequency'],
+                palette=my_pal).set_title("Relative Frequency of Female Pronouns to Total Pronouns")
+    plt.xticks(rotation=90)
+    # plt.show()
+
+    filepng = "visualizations/" + title + ".png"
+    filepdf = "visualizations/" + title + ".pdf"
+    plt.savefig(filepng, bbox_inches='tight')
+    plt.savefig(filepdf, bbox_inches='tight')
+
+    return
+
+
+def bar_sub_obj_freq(she_freq_dict, he_freq_dict, title, x="N/A"):
+    """
+    Creates a bar graph give male/female subject/object frequencies
+    :param she_freq_dict:
+    :param he_freq_dict:
+    :param title: name of the exported file
+    :param x: value of x axis
+    :return:
+    """
+
+    fig, ax = plt.subplots()
+    plt.ylim(0, 1)
+
+    key = []
+
+    for k, v in she_freq_dict.items():
```

Comment:
Maybe try to use more descriptive variable names to make it more readable and help people unfamiliar with what keys and values are associated with she_freq_dict

Reply:
the issue with this is that the key values vary. sometimes it'll be the author gender, other times the date. this function is intended to be run with a data set sorted by the 'freq_by_...' functions. i'll include that in the documentation